analyse_report_data.py

```diff
@@ -0,0 +1,176 @@
+    resources after, boolean indicating if the deletion was completed.
+    """
+    unified = dict()
+    lpreffix = len('freeresources_report_')
+    lsuffix = len('.pickle')
+    names = glob.glob('freeresources_report_*.pickle')
+    names.sort()
+    for name in names:
+        timestamp = name[lpreffix:-lsuffix]
+        users = pickle.load(open(name, 'rb'))
+        for user in users:
+            if user not in users_to_delete:
+                continue
+            if user not in unified:
+                unified[user] = users[user]
+            else:
+                unified[user] = (unified[user][0], users[user][1],
+                                 users[user][2])
+    return unified
+
+
+def print_summary_report(users_to_delete, report):
+    """
+    Print a summary of the report. How many users are in the report in
+    comparison with the ones in the file users_to_delete.txt; how many users'
+    deletions are completed and how many are pending.
+
+    It also prints a summary of the freed resources.
+
+    :param users_to_delete: the list of users to delete, obtained with
+        get_users_to_delete()
+    :param report: the report obtained with get_unified_report()
+    :return: nothing
+    """
+    yes = 0
+    no = 0
+    total_free = dict()
+
+    for value in report.values():
+        if value[2]:
+           yes += 1
+        else:
+           no += 1
+
+        resources_before = value[0]
+        resources_after = value[1]
+        for key in resources_before.keys():
+            removed = len(resources_before[key]) - len(resources_after[key])
+            if removed:
+                total_free[key] = total_free.get(key, 0) + removed
+
+    print '\n===Summary=='
+    print 'Total deleted: ', len(report), '/', len(users_to_delete), \
+        'Completed: ', yes, 'No completed: ', no
+
+    print 'Resources freed: ' + str(total_free)
+
+
+def print_fails_report(report):
+    """Print the users with resources that were not deleted and those
+    resources. Print also a summary of the categories (vms, volumes...) where
+    there are resources not deleted.
+
+    :param report: the report obtained with get_unified_report()
+    :return: nothing
+    """
+
+    print '===Pending deletions report==='
+    categories = set()
+    for key in report:
+        if not report[key][2]:
+            after = report[key][1]
+            modified = dict()
+            for resource in after:
+               if after[resource]:
+                  modified[resource] = after[resource]
+                  categories.add(resource)
+            print key, modified
+
+    print 'Categories not completed: ', categories
+
+
+def save_failed_users_list(users_to_delete, unified):
+    """Create file users_to_delete_failed.txt, with the ids of users who
+    were failed. Only the users with basic type are included.
+
+    A user is considered failed if according the report they have resources
```

Comment:
A user is considered "failed" if he/she has resources after invoking the scripts, or if he/she does not appear in the report, according to the report.